services/src/main/scala/cromwell/pipeline/service/GitLabProjectVersioning.scala

```diff
@@ -235,36 +215,45 @@ class GitLabProjectVersioning(httpClient: HttpClient, config: GitLabConfig)
 
   }
 
-  override def getFile(project: Project, path: Path, version: Option[PipelineVersion])(
-    implicit ec: ExecutionContext
-  ): AsyncResult[ProjectFile] = {
-    val filePath: String = URLEncoderUtils.encode(path.toString)
-    val fileVersion: String = version match {
-      case Some(version) => version.name
-      case None          => getDefaultProjectVersion().toString
+  private def getUpdatedProjectVersion(
+    project: Project,
+    optionUserVersion: Option[PipelineVersion]
+  ): AsyncResult[PipelineVersion] =
+    getLastProjectVersion(project).map(projectVersion => getNewProjectVersion(projectVersion, optionUserVersion))
+
+  private def getLastProjectVersion(project: Project): Future[Option[PipelineVersion]] =
```

Comment:
Quite confusing for me. Is project.version is the same version type that must be returned? If so, why cant we use project.version? How can it be, that passed project has older version. 
If they are different entities, the naming is not very clear.